spring-restdocs-core/src/test/java/org/springframework/restdocs/request/RequestUriSnippetTests.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright 2014-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.restdocs.request;
+
+import java.io.IOException;
+
+import org.junit.Test;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.restdocs.AbstractSnippetTests;
+import org.springframework.restdocs.generate.RestDocumentationGenerator;
+import org.springframework.restdocs.templates.TemplateEngine;
+import org.springframework.restdocs.templates.TemplateFormat;
+import org.springframework.restdocs.templates.TemplateResourceResolver;
+import org.springframework.restdocs.templates.mustache.MustacheTemplateEngine;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.springframework.restdocs.snippet.Attributes.attributes;
+import static org.springframework.restdocs.snippet.Attributes.key;
+
+/**
+ * Tests for {@link RequestUriSnippet}.
+ *
+ * @author Ryan O'Meara
+ */
+public class RequestUriSnippetTests extends AbstractSnippetTests {
+
+	private static final String BOUNDARY = "6o2knFse3p53ty9dmcQvWAIx1zInP11uCfbm";
+
+	public RequestUriSnippetTests(String name, TemplateFormat templateFormat) {
+		super(name, templateFormat);
+	}
+
+	@Test
+	public void getRequest() throws IOException {
+		this.snippets.expectRequestUri().withContents(
+				requestUri(RequestMethod.GET, "/{a}/{b}").header("Accept", "a"));
+
+		new RequestUriSnippet().document(this.operationBuilder
+				.attribute(RestDocumentationGenerator.ATTRIBUTE_NAME_URL_TEMPLATE,
+						"/{a}/{b}")
+				.request("http://localhost/foo/bar").header("Accept", "a").build());
+	}
+
+	@Test
+	public void getRequestWithParameters() throws IOException {
+		this.snippets.expectRequestUri()
+				.withContents(requestUri(RequestMethod.GET, "/{a}/{b}"));
+
+		new RequestUriSnippet().document(this.operationBuilder
+				.attribute(RestDocumentationGenerator.ATTRIBUTE_NAME_URL_TEMPLATE,
+						"/{a}/{b}")
+				.request("http://localhost/foo").header("Alpha", "a").param("b", "bravo")
+				.build());
+	}
+
+	// TODO
```

Comment:
what is the todo?

Reply:
Me not cleaning up properly - it was for writing the tests, which was completed. Thanks for the catch!